src/web_interface/rest/rest_firmware.py

```diff
@@ -106,7 +78,54 @@ def _get_parameters_from_request(request_parameters):
             raise ValueError('Inverted flag can only be used with recursive')
         return query, recursive, inverted, offset, limit
 
-    def _get_with_uid(self, uid):
+    @roles_accepted(*PRIVILEGES['submit_analysis'])
+    @api.expect(firmware_model)
+    def put(self):
+        '''
+        Update existing firmware analysis
+        You can use this endpoint to update a firmware analysis which is already existing
```

Comment:
I think the doc strings of put with and without UID may be switched? (You update a firmware with a UID and upload one without it)